### PR TITLE
Use free instead of realloc to zero size

### PR DESCRIPTION
**Pull request by liu15**

### CODE REVIEW

File: src/chai/ManagedArray_thin.inl [hunk 1]
```diff
@@ -194,16 +194,29 @@ CHAI_HOST void ManagedArray<T>::reallocate(size_t new_elems)
...
+       gpuMemcpy(new_ptr, m_active_pointer, sizeof(T) * m_elems, gpuMemcpyDefault);
+    }
     gpuFree(m_active_pointer);
   #else  // not CHAI_ENABLE_UM
-    new_ptr = static_cast<T*>(realloc(m_active_pointer, sizeof(T) * new_elems));
+    if (new_elems > 0) {
+       new_ptr = static_cast<T*>(realloc(m_active_pointer, sizeof(T) * new_elems));
+    }
+    else {
+       ::free((void *)m_active_pointer);
```
Comment (adayton1) [inline src/chai/ManagedArray_thin.inl:207]: Could set new_ptr to nullptr here, and avoid the changes starting at line 212.

Comment (liu15) [reply src/chai/ManagedArray_thin.inl:207]: Should I set it here plus one for the CHAI_ENABLE_UM branch?  Or give new_ptr a null initializer?  I'm open to anything.... don't know what the best style is here.

Comment (liu15) [reply src/chai/ManagedArray_thin.inl:207]: The changes for CHAI_ENABLE_UM are not strictly required, but it seemed like the analogous change.

Comment (adayton1) [reply src/chai/ManagedArray_thin.inl:207]: I think I would give new_ptr a nullptr initializer.

**Review by adayton1 — APPROVED** [review]
LGTM

**Pull request merged**